tests/unittests/operators/test_logging_utils.py

```
@@ -0,0 +1,112 @@
+        return io.StringIO()
+
+    @pytest.fixture
+    def tee(self, sink):
+        return LineFlushedStdio(sink)
+
+    def test_progress_bar_emits_each_integer_percent(self, tee, sink):
+        for pct in range(0, 101):
+            tee.write(_bar(pct, pct))
+        tee.drain()
+
+        lines = [l for l in sink.getvalue().splitlines() if l.strip()]
```

Ruff (0.15.14)

[error] Ambiguous variable name: `l`

(E741)